PowerShellTools/DebugEngine/ScriptDebugger.cs

```diff
@@ -219,7 +236,11 @@ public void UpdateBreakpoint(DebuggerBreakpointUpdatedEventArgs e)
         /// </summary>
         public void DebuggerFinished()
         {
-            IsDebuggingCommandReady = false;
+            lock (_debuggingCmdLock)
+            {
+                IsDebuggingCommandReady = false;

```

Comment:
Talked offline, this part is just for thread-safe purpose, not specifically for this race condition.